apps/account-worker/src/transaction_publisher/publisher.service.ts

```diff
@@ -114,8 +124,10 @@ export class TransactionPublisherService extends BaseConsumer implements OnAppli
         providerId: job.data.providerId,
         txHash,
         successEvent: targetEvent,
-        birth: tx.era.asMortalEra.birth(lastFinalizedBlockNumber),
-        death: tx.era.asMortalEra.death(lastFinalizedBlockNumber),
+        birth: tx.era.isMortalEra ? tx.era.asMortalEra.birth(lastFinalizedBlockNumber) : lastFinalizedBlockNumber,
+        death: tx.era.isMortalEra
+          ? tx.era.asMortalEra.death(lastFinalizedBlockNumber)
+          : lastFinalizedBlockNumber + IMMORTAL_BLOCK_PERIOD,
```

Comment:
No transactions should be immortal, and I believe I've found the issue. You're trying to get the mortality era of the wrapped transaction, when you really need the era of the `frequencyTxnPayment.payWithCapacity` transaction. `processSingleTransaction` needs to be modified to return the same as `processBatchTransaction`; that is, instead of returning just the hash, it needs to return `Promise<[SubmittableExtrinsic<'promise'>, HexString]>`. See the example of `SIWF_SIGNUP` above for how this works.

Reply:
Good find. Applied the changes and that solved the issue. Thanks